src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -203,6 +204,12 @@ public ODataMessageQuotas MessageQuotas
         /// </summary>
         public Func<string, bool> ShouldIncludeAnnotation { get; set; }
 
+#if NETSTANDARD2_0
+        /// <summary>
+        /// Gets/sets the JSON-CSDL reader settings.
+        /// </summary>
+        public CsdlJsonReaderSettings JsonCsdlReaderSettings { get; set; }
+#endif

```

Comment:
would rather not have a JSON-format specific property on MessageReaderSettings. Is there somewhere we can put this that is specific to the JSON format?

Reply:
Otherwise is to create overload "ReadMetadataDocument" to accept the setting? 

public IEdmModel ReadMetadataDocument(CsdlJsonReaderSettings jsonReaderSetting)

---
In reply to: [518470116](https://github.com/OData/odata.net/pull/1913#discussion_r518470116) [](ancestors = 518470116)

Reply:
I like that slightly better, but it still ties the new overload to JSON.  If we did something like that, I'd rather define a base CsdlReaderSettings that CsdlJsonReaderSettings derived from, and have the new overload take an instance of that base class.  So then if we supported other formats in the future, we could pass the appropriate settings in for that format.

We could even define an XmlReaderSettings that has a Func<uri,xmlreader> property and in 8.0 we can remove the ReadMetadataDocument overload that takes Func<uri,xmlreader>.

---
In reply to: [518471455](https://github.com/OData/odata.net/pull/1913#discussion_r518471455) [](ancestors = 518471455,518470116)

Reply:
Bad news is that we already have "CsdlReaderSettings" and that's for XML specified, sealed. 
When we designed the JSON metadata in OData.Edm, I think we discussed it already.
So, Let me do "public IEdmModel ReadMetadataDocument(CsdlJsonReaderSettings jsonReaderSetting)".

---
In reply to: [518478832](https://github.com/OData/odata.net/pull/1913#discussion_r518478832) [](ancestors = 518478832,518471455,518470116)